certbot-apache/certbot_apache/_internal/configurator.py

```diff
@@ -601,9 +600,9 @@ def _choose_vhost_from_list(self, target_name, temp=False):
                 "in the Apache config.",
                 target_name)
             raise errors.PluginError("No vhost selected")
-        elif temp:
+        if temp:
```

Comment:
The control about `if`/`elif` together with `raise` or `return` is handled as a group with `if`/`else` in the same situation, through the error `no-else-raise`. So disabling one disable all of them.

I think using or not `if`/`else` or `if`/`elif` when a branch interrupts the flow can be more or less relevant depending on the situation. In asymetric branches (eg. one condition is desirable, the others are not and trigger an error flow), not using `else` or `if` makes more sense. In symetric conditions, it is a little clearer to use these keywords.

A good article about this: https://dmerej.info/blog/post/else-after-return-yea-or-nay/

If I need to choose between the two rules, I would still keep `no-else-raise`. Because from my experience on this language and others, the asymetric branches occurs more often, and in that specific case, using `else` tends to create nested branches that harm readability.

Reply:
Isn't the error the various `no-else-*` described at http://pylint.pycqa.org/en/latest/technical_reference/features.html#refactoring-checker-messages?

I agree which style is better depends on the situation, but pylint unconditionally errors meaning the times the programmer thinks the if/elif/else version is better tests will fail and then they either have to refactor their code to appease pylint when they disagree or add more `pylint: disable` comments. I think this is annoying, wastes time, results in uglier code in the case of symmetric branches, feels like you're fighting our tools, and think this is better left up to the code's author and reviewer.

I still think these checks should be disabled. I'd prefer if you reverted the code changes to appease this `pylint` error too.

Reply:
I agree it should be disabled, and push the design decision to the author or the reviewer.

I disagree about reverting the changes on asymetric branches, as it is generally recognized that it makes the code clearer.